examples/optuna/skorch_example.py

```diff
@@ -0,0 +1,100 @@
+import argparse
+import urllib
+
+import numpy as np
+import optuna
+from optuna.integration import SkorchPruningCallback
+import skorch
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import pandas as pd
+
+from sklearn.datasets import fetch_openml
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+
+
+SUBSET_RATIO = 0.4
+
+mnist = fetch_openml("mnist_784", cache=False)
+
+X = pd.DataFrame(mnist.data)
+y = mnist.target.astype("int64")
+indices = np.random.permutation(len(X))
+N = int(len(X) * SUBSET_RATIO)
+X = X.iloc[indices][:N].astype(np.float32)  
+y = y[indices][:N]
+
+X /= 255.0
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=42)
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+class ClassifierModule(nn.Module):
+    def __init__(self, trial: optuna.Trial) -> None:
+        super().__init__()
+
+        # We optimize the number of layers, hidden units in each layer and dropouts.
+        layers = []
+        n_layers = trial.suggest_int("n_layers", 1, 3)
+        dropout = trial.suggest_float("dropout", 0.2, 0.5)
+        input_dim = 28 * 28
+        for i in range(n_layers):
+            output_dim = trial.suggest_int("n_units_l{}".format(i), 4, 128, log=True)
+            layers.append(nn.Linear(input_dim, output_dim))
+            layers.append(nn.Dropout(dropout))
+            layers.append(nn.ReLU())
+            input_dim = output_dim
+
+        layers.append(nn.Linear(input_dim, 10))
+
+        self.model = nn.Sequential(*layers)
+
+    def forward(self, x):
+        if isinstance(x, dict):
+            x = x["data"] 
+        return F.softmax(self.model(x), dim=-1)
```

Comment:
Instead of calling `F.softmax` here, it's easier to append an `nn.Softmax(dim=-1)` layer to the `layers` list during `__init__`.